infrastructure/src/main/java/org/corfudb/infrastructure/management/FullyConnectedClusterRecommendationEngine.java

```diff
@@ -0,0 +1,391 @@
+package org.corfudb.infrastructure.management;
+
+import org.corfudb.protocols.wireprotocol.PeerView;
+import org.corfudb.runtime.view.Layout;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * This class is an implementation of {@link ClusterRecommendationPolicy} where the ideal state
+ * of the Corfu cluster is a fully connected network (i.e. complete graph) in which there is an
+ * active link amongst all nodes in the cluster. Failed and Healed nodes are recommended so that
+ * the cluster remains fully connected.
+ *
+ * Created by Sam Behnam on 10/27/18.
+ */
+public class FullyConnectedClusterRecommendationEngine implements ClusterRecommendationEngine {
```

Comment:
I actually like this, it cleans up the current code and it is modular enough to be easily tested. Can you add tests?

Reply:
Sounds good. yes will do.